### ISSUE TITLE
[TW-590] task rc.color:off ghistory{,.monthly}

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#625]
_Uli Martens on 2012-02-12T10:05:44Z says:_ 

 <pre>
$ task show alias.ghistory

Config Variable Value           
alias.ghistory  ghistory.monthly
[..]
$ task rc.color:off ghistory
[output with colors]
$ task rc.color:off ghistory.monthly
[output with ascii-art]
</pre>

adding this in  src/commands/CmdHistory.cpp 371ff shows that context.color actually is set for the first command, and cleared for the second
<pre>
fprintf(stderr, "am in CmdGHistoryMonthly\n");             
fprintf(stderr, "context.color is: %i\n", context.color());
</pre>

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2012-02-12T10:05:44Z
Modified: 2014-02-09T02:06:29Z
```

**Comment by taskwarrior** [comment]
_Uli Martens on 2012-02-12T10:13:36Z says:_ 

 the alias expansion seems to lose all config overrides, (at)task rc.debug:1 rc.color:off ghistory@ is executed as if it was (at)task ghistory@...

**Comment by taskwarrior** [comment]
_Uli Martens on 2012-02-12T11:40:14Z says:_ 

 A3::resolve_aliases() re-added the already parsed values to the argument vector when it expanded an alias but lost categorization. Then A3::apply_overrides() scanned for those arguments categorized as overrides - but all categorization was lost in the step before...

Attached patch fixes this problem.

**Comment by taskwarrior** [comment]
_Uli Martens on 2012-02-12T13:20:05Z says:_ 

 imported/closed in 41cc1f9

**Comment by pbeckingham** [comment]
[TW-590_categorize.diff.txt](https://github.com/GothenburgBitFactory/taskwarrior/files/1756179/TW-590_categorize.diff.txt)
